src/connection.rs

```diff
@@ -427,44 +423,28 @@ impl ConnectionLike for Connection {
 /// }
 /// # }
 /// ```
-impl PubSub {
-    fn get_channel<T: ToRedisArgs>(&mut self, channel: &T) -> Vec<u8> {
-        let mut chan = vec![];
-        for item in channel.to_redis_args().iter() {
-            chan.extend(item.iter().cloned());
-        }
-        chan
-    }
-
+impl<'a> PubSub<'a> {
     /// Subscribes to a new channel.
     pub fn subscribe<T: ToRedisArgs>(&mut self, channel: T) -> RedisResult<()> {
-        let chan = self.get_channel(&channel);
-        let _: () = try!(cmd("SUBSCRIBE").arg(&*chan).query(&self.con));
-        self.channels.insert(chan);
+        let _: () = try!(cmd("SUBSCRIBE").arg(channel).query(self.con));
```

Comment:
While we're at it, can we switch to the postfix question mark operator here (and below)?

Reply:
Yep, happy to do so. I believe I went with `try!` to be consistent with the rest of the code base.